src/plugins/query_enhancements/server/routes/data_source_connection/routes.ts

```diff
@@ -99,4 +99,31 @@ export function registerDataSourceConnectionsRoutes(
       }
     }
   );
+
+  router.delete(
+    {
+      path: `${API.DATA_SOURCE.ASYNC_JOBS}`,
+      validate: {
+        query: schema.object({
+          id: schema.string(),
```

Comment:
nit. is it conventional to use `id` for datasource id? can this be `datasourceId` to avoid confusion since there's also `queryId` below?

Reply:
Was just following the rest of the file to add in parameters, but I agree that the rest of query enhancements routes + APIs should update this. This would be out of scope for this PR though, and should be a followup item.